### ISSUE TITLE
Console logs are generated from instrument.js file

### ISSUE THREAD

**Issue by harshalijain** [getsentry/sentry-react-native#794]
OS:

- MacOS

Platform:
- iOS
- Android

SDK:
- `@sentry/react-native`

SDK version: 1.2.0
`react-native` version: 0.59.10

Are you using Expo?
- No

Are you using sentry.io or on-premise?
-  sentry.io (SaaS)

---
I have following issue:

In my system, while debugging on Chrome , all my logs are coming from a file named instrument.js
which is placed at  /node_modules/@sentry/utils/dist/instrument.js at line 104
But I want my logs to show the line from where log is originated.

Let me know if there is any possible configuration that I am missing.

**Comment by HazAT** [comment]
Hey, this should solve your issue: https://github.com/getsentry/sentry-javascript/issues/1003#issuecomment-380310913

**Comment by harshalijain** [comment]
Tried above fix, didn't work, I had to comment the line in instrument.js file from which all the logs were generated.

**Comment by YisusMB** [comment]
please re-open this issue, i have the same problem all console logs shows instrument.js 112
<img width="864" alt="Captura de Pantalla 2020-03-17 a la(s) 12 34 12" src="https://user-images.githubusercontent.com/43631579/76889667-9df2ea80-684b-11ea-89ea-624967054223.png">

**Comment by harshalijain** [comment]
In instrument.js file, 
do the following changes in the function containing this line points to :

```
function instrumentConsole() {
    if (('console' in global)) {
        return;
    }
   
}
```

Hence, if there is a console module defined globally, it will return, we just have to reverse the condition. However, this change needs to be reverted in production build or hence console logs might not get sent.

**Comment by simonarcher** [comment]
None of the above are valid solutions.. can this please be re-opened?

**Comment by HazAT** [comment]
OK sorry if I wasn't clear enough, adding `/ravenjs/` is wrong you need to add `/@sentry/` then it works just fine.

Before:
![image](https://user-images.githubusercontent.com/363802/78777448-d5ac0880-7999-11ea-949a-f94ed61dd968.png)


After:
![image](https://user-images.githubusercontent.com/363802/78777397-bdd48480-7999-11ea-986e-fab238bc6e04.png)


Setting:
![image](https://user-images.githubusercontent.com/363802/78777477-e492bb00-7999-11ea-913f-c4bbf4fa1175.png)

**Comment by JuraJuki** [comment]
This solution is not working. Tried restarting Chrome after adding the blackbox, it didn't help

**Comment by ezequieljardim** [comment]
> This solution is not working. Tried restarting Chrome after adding the blackbox, it didn't help

Try adding `instrument.ts` to the Blackbox. It just worked for me

**Comment by elkwood** [comment]
Ignoring script in Firefox does not change the behavior. This issue makes debugging useless since the console.log stack is completely gone.

**Comment by UnderscoreShift** [comment]
Be sure to add **both** `instrument.ts` and `/@sentry/` to Blackboxing.

**Comment by sshadmand** [comment]
Echo @elkwood , Firefox does not seem to have a way to work around this. I can ignore the `instrument.ts` file but there is no option to Blackbox `/@sentry/`. @UnderscoreShift screenshots for FF?
<img width="786" alt="Screen Shot 2021-08-17 at 8 15 21 PM" src="https://user-images.githubusercontent.com/68735/129816835-d607f71d-2077-49fb-9a1b-277aaeeb4d10.png">

**Comment by barthicus** [comment]
I fixed this by setting `console` field to `false` in the Breadcrumbs Integration which is one of the default Sentry Integrations.
Docs: https://docs.sentry.io/platforms/javascript/configuration/integrations/default/#breadcrumbs

In my case I had to add an integration config field:

```
Sentry.init({
  ...initOptions,
  integrations: [
    new Sentry.Integrations.Breadcrumbs({
      console: false
    })
  ],
  tracesSampler: () => (process.env.NODE_ENV === 'production' ? 1 : 0.2)
})
```

I pasted my Sentry init example for the Next.js project but it should be similar in other similar environments.

**Comment by AlaaEl-DinAhmed** [comment]
> I fixed this by setting `console` field to `false` in the Breadcrumbs Integration which is one of the default Sentry Integrations. Docs: https://docs.sentry.io/platforms/javascript/configuration/integrations/default/#breadcrumbs
> 
> In my case I had to add an integration config field:
> 
> ```
> Sentry.init({
>   ...initOptions,
>   integrations: [
>     new Sentry.Integrations.Breadcrumbs({
>       console: false
>     })
>   ],
>   tracesSampler: () => (process.env.NODE_ENV === 'production' ? 1 : 0.2)
> })
> ```
> 
> I pasted my Sentry init example for the Next.js project but it should be similar in other similar environments.

Thank you so much, your solution works fine with me

**Comment by amritk** [comment]
> I fixed this by setting `console` field to `false` in the Breadcrumbs Integration which is one of the default Sentry Integrations. Docs: https://docs.sentry.io/platforms/javascript/configuration/integrations/default/#breadcrumbs
> 
> In my case I had to add an integration config field:
> 
> ```
> Sentry.init({
>   ...initOptions,
>   integrations: [
>     new Sentry.Integrations.Breadcrumbs({
>       console: false
>     })
>   ],
>   tracesSampler: () => (process.env.NODE_ENV === 'production' ? 1 : 0.2)
> })
> ```
> 
> I pasted my Sentry init example for the Next.js project but it should be similar in other similar environments.

Wont that stop passing the console.logs into sentry?

**Comment by meetzaveri** [comment]
For me, adding `instrument.ts` to "Ignore Lists" in chrome devtools settings worked.

**Comment by dannydi12** [comment]
> > I fixed this by setting `console` field to `false` in the Breadcrumbs Integration which is one of the default Sentry Integrations. Docs: https://docs.sentry.io/platforms/javascript/configuration/integrations/default/#breadcrumbs
> > In my case I had to add an integration config field:
> > ```
> > Sentry.init({
> >   ...initOptions,
> >   integrations: [
> >     new Sentry.Integrations.Breadcrumbs({
> >       console: false
> >     })
> >   ],
> >   tracesSampler: () => (process.env.NODE_ENV === 'production' ? 1 : 0.2)
> > })
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I pasted my Sentry init example for the Next.js project but it should be similar in other similar environments.
> 
> Wont that stop passing the console.logs into sentry?

I was wondering the same thing... @barthicus is this the case?

**Comment by barthicus** [comment]
> > > I fixed this by setting `console` field to `false` in the Breadcrumbs Integration which is one of the default Sentry Integrations. Docs: https://docs.sentry.io/platforms/javascript/configuration/integrations/default/#breadcrumbs
> > > In my case I had to add an integration config field:
> > > ```
> > > Sentry.init({
> > >   ...initOptions,
> > >   integrations: [
> > >     new Sentry.Integrations.Breadcrumbs({
> > >       console: false
> > >     })
> > >   ],
> > >   tracesSampler: () => (process.env.NODE_ENV === 'production' ? 1 : 0.2)
> > > })
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I pasted my Sentry init example for the Next.js project but it should be similar in other similar environments.
> > 
> > 
> > Wont that stop passing the console.logs into sentry?
> 
> I was wondering the same thing... @barthicus is this the case?

@dannydi12 @amritk  Hmm, console logs by default shouldn't be send to Sentry AFAIK.
As I can see in docs in order to enable sending logs when `console.log` occurs you would have to implement the **CaptureConsole** integration first: https://docs.sentry.io/platforms/javascript/configuration/integrations/plugin/#captureconsole

Then you can set which type of logs you want to send to Sentry.
Even if you set both integrations they shouldn't collide with each other. IMO disabling console.log to browser's console via **Breadcrumbs** integration shouldn't conflict with sending logs to Sentry via `console.logs`.

But I'm casual Sentry user, I used it only once some time ago and I'm not that familiar with the whole product.
Maybe someone more experienced could confirm my words 😉
Cheers!

**Comment by Bartel-C8** [comment]
I have the same issue in the @sentry/react project.
Console.logs are unusable to browse to sources.

But maybe the console.logs are caught by default there? (As actually it is quite useful, because the console.logs could help debugging). But it makes the source navigation impossible (without manually adding sentry to the ignore/filter list).

**Comment by marandaneto** [comment]
In case you're facing new issues, please raise a new issue with more context, the solution was already posted as a comment before, thanks.
